CRM/Logging/Differ.php

```diff
@@ -416,6 +420,12 @@ public function getAllChangesForConnection($tables) {
         $sql .= " UNION SELECT '{$table}' as table_name, id FROM {$this->db}.log_{$table} WHERE log_conn_id = %1";
       }
     }
+    if ($limit) {
+      $sql .= " LIMIT {$limit}";
```

Comment:
I can see this gets escaped in another place lower down if it comes thru the url, but I'd argue that since (a) this is a standalone public function that can take arbitrary params, and (b) somebody might refactor the other place later removing the protection, that it should escape it here. It can still call escape in the other place too - it would fatal out at that point if it's not even an int.

Reply:
That's an excellent valid point, I'll do the changes right away.